### ISSUE TITLE
Left Join with Scio SQL leads to NPE 

### ISSUE THREAD

**Issue by anish749** [spotify/scio#1953]
Using a similar setup as #1951 

The following works
```
 tsql"SELECT s.userId, s.userIdLong, s.fields FROM $s s INNER JOIN $a ON $s.userId = $a.artistId"
.as[User]
```

Using a Left Join 
```
 tsql"SELECT s.userId, s.userIdLong, s.fields FROM $s s LEFT JOIN $a ON $s.userId = $a.artistId"
.as[User]
```
we get a NPE:
```

Exception in thread "main" org.apache.beam.sdk.Pipeline$PipelineExecutionException: java.lang.NullPointerException
	at org.apache.beam.runners.direct.DirectRunner$DirectPipelineResult.waitUntilFinish(DirectRunner.java:332)
	at org.apache.beam.runners.direct.DirectRunner$DirectPipelineResult.waitUntilFinish(DirectRunner.java:302)
	at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:197)
	at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:64)
	at org.apache.beam.sdk.Pipeline.run(Pipeline.java:313)
	at org.apache.beam.sdk.Pipeline.run(Pipeline.java:299)
	at com.spotify.scio.ScioContext$$anonfun$close$1.apply(ScioContext.scala:570)
	at com.spotify.scio.ScioContext$$anonfun$close$1.apply(ScioContext.scala:558)
	at com.spotify.scio.ScioContext.requireNotClosed(ScioContext.scala:590)
	at com.spotify.scio.ScioContext.close(ScioContext.scala:558)
	at qscio.DataflowBenchmark.benchLocal(DataflowBenchmark.scala:69)
	at qscio.DataflowBenchmark.benchLocal$(DataflowBenchmark.scala:63)
	at benchmark.ExpressionJoin$.benchLocal(ExpressionJoin.scala:7)
	at benchmark.ExpressionJoin$.main(ExpressionJoin.scala:8)
	at benchmark.ExpressionJoin.main(ExpressionJoin.scala)
Caused by: java.lang.NullPointerException
	at org.apache.beam.sdk.values.Row$Equals.deepHashCodeForList(Row.java:469)
	at org.apache.beam.sdk.values.Row$Equals.deepHashCode(Row.java:398)
	at org.apache.beam.sdk.values.Row.hashCode(Row.java:369)
	at java.util.Arrays.deepHashCode(Arrays.java:4207)
	at org.apache.beam.sdk.values.KV.hashCode(KV.java:121)
	at java.util.Arrays.deepHashCode(Arrays.java:4207)
	at org.apache.beam.sdk.values.KV.hashCode(KV.java:121)
	at java.util.Arrays.hashCode(Arrays.java:4146)
	at java.util.Objects.hash(Objects.java:128)
	at org.apache.beam.sdk.util.WindowedValue$ValueInGlobalWindow.hashCode(WindowedValue.java:245)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:557)
	at org.apache.beam.vendor.guava.v20_0.com.google.common.collect.AbstractMapBasedMultimap.put(AbstractMapBasedMultimap.java:191)
	at org.apache.beam.vendor.guava.v20_0.com.google.common.collect.AbstractSetMultimap.put(AbstractSetMultimap.java:130)
	at org.apache.beam.vendor.guava.v20_0.com.google.common.collect.HashMultimap.put(HashMultimap.java:48)
	at org.apache.beam.runners.direct.ImmutabilityCheckingBundleFactory$ImmutabilityEnforcingBundle.add(ImmutabilityCheckingBundleFactory.java:111)
	at org.apache.beam.runners.direct.ParDoEvaluator$BundleOutputManager.output(ParDoEvaluator.java:264)
	at org.apache.beam.repackaged.beam_runners_direct_java.runners.core.SimpleDoFnRunner.outputWindowedValue(SimpleDoFnRunner.java:247)
	at org.apache.beam.repackaged.beam_runners_direct_java.runners.core.SimpleDoFnRunner.access$700(SimpleDoFnRunner.java:73)
	at org.apache.beam.repackaged.beam_runners_direct_java.runners.core.SimpleDoFnRunner$DoFnProcessContext.output(SimpleDoFnRunner.java:559)
	at org.apache.beam.repackaged.beam_runners_direct_java.runners.core.SimpleDoFnRunner$DoFnProcessContext.output(SimpleDoFnRunner.java:547)
	at org.apache.beam.sdk.extensions.joinlibrary.Join$LeftOuterJoin$1.processElement(Join.java:154)

Process finished with exit code 1

```

There isn't a lot of hints about where the error is.

**Comment by regadas** [comment]
This one I would replicate! Thx for reporting!

**Comment by regadas** [comment]
This is actually seems a bug in beam https://github.com/apache/beam/pull/8368. Fixed in `2.13.0` and with #1952

**Comment by anish749** [comment]
Thanks for the quick response and the version upgrade. :)
